### PR TITLE
add option to extend problem json error

### PR DESCRIPTION
**Pull request by blankdots**
### Description

add additional members to problem json response if the content type can be converted to dict
### Related issues

related for #641 and required by some dev in #633 

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

- update pre-commit dependencies, as packages were updated this month, so that we keep in sync
- add dict members to `application/problem+json` if that exist in the HTTP exception

all that needs to be added is:
```
text=ujson.dumps({"coolContent": [{"value": 1}, {"value": 2}]}),
content_type="application/json"
```

### Testing

- [x] Needs testing (start an issue or do a follow up PR about it)

### CODE REVIEW

File: metadata_backend/api/middlewares.py [hunk 1]
```diff
@@ -114,16 +119,25 @@ def _json_problem(exception: web.HTTPError, url: URL, _type: str = "about:blank"
...
+        # a custom error document when one exists
+        "type": _type,
+        "title": HTTPStatus(exception.status).phrase,
+        "detail": exception.reason,
+        "status": exception.status,
+        "instance": url.path,  # optional
+    }
+    # we require the additional members to be sent as dict
+    # so that we can easily append them to preformated response
+    if exception.text != exception.reason and exception.content_type == "application/json":
```
Comment (csc-felipe) [inline metadata_backend/api/middlewares.py:133]: Nice hack

File: metadata_backend/api/middlewares.py [hunk 2]
```diff
@@ -100,7 +100,12 @@ async def check_session(req: web.Request, handler: aiohttp_session.Handler) -> w
 
 
 def _unauthorized(reason: str) -> web.HTTPUnauthorized:
-    return web.HTTPUnauthorized(headers={"WWW-Authenticate": 'OAuth realm="/", charset="UTF-8"'}, reason=reason)
+    return web.HTTPUnauthorized(
+        headers={"WWW-Authenticate": 'OAuth realm="/", charset="UTF-8"'},
+        reason=reason,
+        text=ujson.dumps({"coolContent": [{"value": 1}, {"value": 2}]}),
```
Comment (csc-felipe) [inline metadata_backend/api/middlewares.py:106]: I'm guessing this shouldn't be here ^^

**Review by csc-jm — APPROVED** [review]

File: metadata_backend/api/middlewares.py (hunk 2 above)
Comment (blankdots) [reply metadata_backend/api/middlewares.py:106]: yeah, my bad, was testing

File: metadata_backend/api/middlewares.py (hunk 1 above)
Comment (blankdots) [reply metadata_backend/api/middlewares.py:133]: yeah, i recognize, could be better, but these should be enough for current needs.

**Review by csc-jm — APPROVED** [review]

**Review by csc-felipe — APPROVED** [review]

**Pull request merged**